src/m365/spo/commands/file/file-sharinglink-set.ts

```diff
@@ -0,0 +1,221 @@
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '-i, --fileId [fileId]'
+      },
+      {
+        option: '-f, --fileUrl [fileUrl]'
+      },
+      {
+        option: '--id <id>'
+      },
+      {
+        option: '--role [role]',
+        autocomplete: SpoFileSharingLinkSetCommand.role
+      },
+      {
+        option: '--expirationDateTime [expirationDateTime]'
+      },
+
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (args.options.fileId && !validation.isValidGuid(args.options.fileId)) {
+          return `${args.options.fileId} is not a valid GUID`;
+        }
+
+        if (args.options.id && !validation.isValidGuid(args.options.id)) {
+          return `${args.options.id} is not a valid GUID`;
+        }
+
+        if (args.options.role &&
+          SpoFileSharingLinkSetCommand.role.indexOf(args.options.role) < 0) {
+          return `'${args.options.role}' is not a valid scope type. Allowed scope types are ${SpoFileSharingLinkSetCommand.role.join(', ')}`;
+        }
+
+        const parsedDateTime = Date.parse(args.options.expirationDateTime as string);
+        if (args.options.expirationDateTime && !(!parsedDateTime) !== true) {
+          return `${args.options.expirationDateTime} is not a valid date format. Provide the date in one of the following formats:
+      ${chalk.grey('YYYY-MM-DD')}
+      ${chalk.grey('YYYY-MM-DDThh:mm')}
+      ${chalk.grey('YYYY-MM-DDThh:mmZ')}
+      ${chalk.grey('YYYY-MM-DDThh:mm±hh:mm')}`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['fileId', 'fileUrl'] });
+    this.optionSets.push({ options: ['role', 'expirationDateTime'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Updates a sharing link for file ${args.options.fileId || args.options.fileUrl}...`);
+    }
+
+    try {
+      const fileDetails = await this.getFileDetails(args.options.webUrl, args.options.fileId, args.options.fileUrl);
+      const sharingInformation = await this.getSharingInformation(args.options.webUrl, fileDetails, args.options.id);
+      if (args.options.role) {
+        sharingInformation.role = args.options.role === "read" ? 1 : 2;
+      }
+      else {
+        sharingInformation.expiration = this.parseDateExact(args.options.expirationDateTime);
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${args.options.webUrl}/_api/web/Lists(@a1)/GetItemByUniqueId(@a2)/ShareLink?@a1='${fileDetails.ListId}'&@a2='${fileDetails.UniqueId}'`,
```

Comment:
If the provided Graph endpoint doesn't support what the spec says, I think we should discuss it in the spec itself. Now, we are using an endpoint that is different from all other `file sharinglink` commands. While this works for now, we might run into issues later on (they are gradually adding more features to the sharing link Graph API). Therefore I suggest that we still use the Graph API instead of this one. If the Graph API doesn't support to change the role, then we shouldn't provide this option.

Reply:
since we remove the `role` option, should `expirationDateTime` be made required?

Reply:
Good question. Since it's the only option left, I'd say yes, let's make it required.